colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -117,29 +121,72 @@ class Collection(val config: CollectorConfig) {
         )
       }
     }
-    if (collectors.containsKey(address)) {
-      cast(collectors.get(address))
-    } else {
+    addContext(ns)
```

Comment:
At the very least, log something, but yea it is a bit of an edgy case.   I'm not strongly opinionated on any 3(unless we overwrite, then we should log something at debug? level)